Python/pytime.c

```diff
@@ -1004,29 +1008,37 @@ py_get_system_clock(PyTime_t *tp, _Py_clock_info_t *info, int raise_exc)
 }
 
 
-PyTime_t
-_PyTime_TimeUnchecked(void)
+int
+PyTime_Time(PyTime_t *result)
 {
-    PyTime_t t;
-    if (py_get_system_clock(&t, NULL, 0) < 0) {
-        // If clock_gettime(CLOCK_REALTIME) or gettimeofday() fails:
-        // silently ignore the failure and return 0.
-        t = 0;
+    if (py_get_system_clock(result, NULL, 1) < 0) {
+        *result = 0;
+        return -1;
     }
-    return t;
+    return 0;
 }
 
 
 int
-PyTime_Time(PyTime_t *result)
+PyTime_TimeRaw(PyTime_t *result)
 {
-    if (py_get_system_clock(result, NULL, 1) < 0) {
+    if (py_get_system_clock(result, NULL, 0) < 0) {
         *result = 0;
         return -1;
     }
     return 0;
 }
 
+
+PyTime_t
+_PyTime_TimeUnchecked(void)
+{
+    PyTime_t t;
+    (void)PyTime_TimeRaw(&t);
```

Comment:
We can ensure that getting the time *is* infallible on current platforms. A debug-build check seems like the best tool:
```suggestion
    int result = PyTime_TimeRaw(&t);
    assert(result);
```

(Note that a `_PyDeadline_Init`-based mutex timeout -- our main use for the internal Unchecked API -- will *deadlock* if `_PyTime_MonotonicUnchecked` always returns 0.)